c_cxx/Snpe_EP/README.md

```diff
@@ -0,0 +1,110 @@
+# Image classification with Inception v3 in CPP using SNPE Execution Provider
+1.  The imagel classification uses a Inception v3 model from Snpe turtorials with DLC format. Wrap it as a Onnx model in Onnx model to inference with SNPE Execution Provider.
+2.  The sample uses the Onnxruntime SNPE Execution Provider to run inference on various Qualcomm devices like Qualcomm CPU, GPU DSP, AIP, etc. It supports Windows ARM64, and Android.
+
+# Prerequisites (Linux host)
+1. Get a Linux host
+2. Download SNPE SDK from [here](https://developer.qualcomm.com/software/qualcomm-neural-processing-sdk)
+3. Setup SNPE on the Linux host. Setup environment for the tutorial. Follow the Tutorials and Examples for [Tensorflow inceptionv3](https://developer.qualcomm.com/sites/default/files/docs/snpe/tutorial_inceptionv3.html)
+4. Get the model [Inception v3](https://developer.qualcomm.com/sites/default/files/docs/snpe/tutorial_setup.html#tutorial_setup_inception_v3)
+
+    ```
+    python $SNPE_ROOT/models/inception_v3/scripts/setup_inceptionv3.py -a ~/tmpdir -d -r dsp
+    ```
+
+    The generated DLC file inception_v3.dlc and inception_v3_quantized.dlc can be found at $SNPE_ROOT/models/inception_v3/dlc/.
+
+    The data chairs.raw, notice_sign.raw, plastic_cup.raw and trash_bin.raw can be found at $SNPE_ROOT/models/inception_v3/data/. The sample applicatioin use these raw file as input.
+
+5. Create Onnx model from DLC file
+
+    Create a script gen_onnx_model.py in folder $SNPE_ROOT/models/inception_v3 with the code below:
+
+    ```
+    import onnx
+    from onnx import helper
+    from onnx import TensorProto
+
+    with open('./dlc/inception_v3_quantized.dlc','rb') as file:
+        file_content = file.read()
+
+    input1 = helper.make_tensor_value_info('input:0', TensorProto.FLOAT, [1, 299, 299, 3])
+    output1 = helper.make_tensor_value_info('InceptionV3/Predictions/Reshape_1:0', TensorProto.FLOAT, [1, 1001])
+
+    snpe_node = helper.make_node('Snpe', name='Inception v3', inputs=['input:0'], outputs=['InceptionV3/Predictions/Reshape_1:0'], DLC=file_content, snpe_version='1.61.0', target_device='DSP', notes='quantized dlc model.', domain='com.microsoft')
+
+    graph_def = helper.make_graph([snpe_node], 'Inception_v3', [input1], [output1])
+    model_def = helper.make_model(graph_def, producer_name='tesorflow', opset_imports=[helper.make_opsetid('', 13)])
+    onnx.save(model_def, 'snpe_inception_v3.onnx')
+    ```
+
+    Run the script to generate the Onnx model with the DLC content embed in a Onnx node.
+    Change the data type for input & output from TensorProto.FLOAT to TensorProto.UINT8 if you want to run model inference with quantized data. It's helpful to save the bandwidth (expecially for application with CS mode, data need to be transmitted across the network). Also need to change the options_values for buffer_type from FLOAT to TF8 in the main.cpp

```

Comment:
Is the choice for the data type required to match the `provider_options` in the EP?

Reply:
that's what I was trying to say, let me make it clear.